src/storage/conversion/ConversionUtil.ts

```diff
@@ -13,51 +12,49 @@ import type { RepresentationConverterArgs } from './RepresentationConverter';
  * Since more specific media ranges override less specific ones,
  * this will be ignored if there is a specific internal type preference.
  *
- * @param preferences - Preferences for output type.
- * @param types - Media types to compare to the preferences.
+ * @param preferredTypes - Preferences for output type.
+ * @param availableTypes - Media types to compare to the preferences.
  *
  * @throws BadRequestHttpError
  * If the type preferences are undefined or if there are duplicate preferences.
  *
  * @returns The weighted and filtered list of matching types.
  */
-export const matchingMediaTypes = (preferences: RepresentationPreferences, types: string[]):
-RepresentationPreference[] => {
-  if (!Array.isArray(preferences.type)) {
-    throw new BadRequestHttpError('Output type required for conversion.');
-  }
-
-  const prefMap = preferences.type.reduce((map: Record<string, number>, pref): Record<string, number> => {
-    if (map[pref.value]) {
-      throw new BadRequestHttpError(`Duplicate type preference found: ${pref.value}`);
-    }
-    map[pref.value] = pref.weight;
-    return map;
-  }, {});
-
+export const matchingMediaTypes = (preferredTypes: ValuePreferences = {}, availableTypes: ValuePreferences = {}):
+string[] => {
+  // No preference means anything is acceptable
+  const preferred = { ...preferredTypes };
+  if (Object.keys(preferredTypes).length === 0) {
+    preferred['*/*'] = 1;
   // Prevent accidental use of internal types
-  if (!prefMap[INTERNAL_ALL]) {
-    prefMap[INTERNAL_ALL] = 0;
+  } else if (!(INTERNAL_ALL in preferred)) {
+    preferred[INTERNAL_ALL] = 0;
   }
 
   // RFC 7231
   //    Media ranges can be overridden by more specific media ranges or
   //    specific media types.  If more than one media range applies to a
   //    given type, the most specific reference has precedence.
-  const weightedSupported = types.map((type): RepresentationPreference => {
+  const weightedSupported = Object.entries(availableTypes).map(([ type, quality ]): [string, number] => {
     const match = /^([^/]+)\/([^\s;]+)/u.exec(type);
     if (!match) {
       throw new InternalServerError(`Unexpected type preference: ${type}`);
     }
     const [ , main, sub ] = match;
-    const weight = prefMap[type] ?? prefMap[`${main}/${sub}`] ?? prefMap[`${main}/*`] ?? prefMap['*/*'] ?? 0;
-    return { value: type, weight };
+    const weight =
+      preferred[type] ??
+      preferred[`${main}/${sub}`] ??
+      preferred[`${main}/*`] ??
+      preferred['*/*'] ??
+      0;
+    return [ type, weight * quality ];
```

Comment:
@joachimvh This single line is the point of this entire PR.

Reply:
If that's the only line you wanted to add you could have used a constant 😉